src/app/store/actions/product.actions.ts

```diff
@@ -0,0 +1,74 @@
+import {createAction, props} from "@ngrx/store";
+import {Product} from "../../model/product";
+
+export const loadProducts = createAction('[ProductList - Page] Load Todos');
+
+export const loadProductsSuccess = createAction(
+  '[Product API] Product Load Success',
+  props<{ products: Product[] }>()
+);
+
+export const loadProductsFailure = createAction(
```

Comment:
I advise you to make only one failure action for everything, unless you explicitly need individual ones. As far as I see, you only send the error, so it's the same thing repeated over and over again.